botocore/signers.py

```diff
@@ -130,3 +134,275 @@ def get_auth(self, signing_name, region_name, signature_version=None):
+    :param client: The client that will be used to generate the url
+
+    :type bucket: string
+    :param bucket: The name of the bucket to presign the post to. Note that
+        bucket related conditions should not be included in the
+        ``conditions`` parameter.
+
+    :type key: string
+    :param key: Key name, optionally add ${filename} to the end to
+        attach the submitted filename. Note that key related condtions and
+        fieldss are filled out for you and should not be included in the
+        ``fields`` or ``condtions`` parmater.
+
+    :type fields: dict
+    :param fields: A dictionary of prefilled form fields to build on top
+        of. Elements that may be included are acl, Cache-Control,
+        Content-Type, Content-Disposition, Content-Encoding, Expires,
+        success_action_redirect, redirect, success_action_status,
+        and x-amz-meta-.
+
+        Note that if a particular element is included in the fields
+        dictionary it will not be automatically added to the conditions list.
+        You must specify a condition for the element as well.
+
+    :type conditions: list
+    :param conditions: A list of conditions to include in the policy. Each
+        element can be either a list or a structure. For example:
+
+        [
+         {"acl": "public-read"},
+         ["content-length-range", 2, 5],
+         ["starts-with", "$success_action_redirect", ""]
+        ]
+
+        Conditions that are included may pertain to acl, content-length-range,
+        Cache-Control, Content-Type, Content-Disposition, Content-Encoding,
+        Expires, success_action_redirect, redirect, success_action_status,
+        and/or x-amz-meta-.
+
+        Note that if you include a condition, you must specify
+        the a valid value in the fields dictionary as well. A value will not
+        be added automatically to the fields dictionary based on the
+        conditions.
+
+    :type expires_in: int
+    :param expires_in: The number of seconds the presigned post is valid for.
+
+    :rtype: dict
+    :returns: A dictionary with two elements: ``url`` and ``fields``.
+        Url is the url to post to. Fields is a dictionary filled with
+        the form fields and respective values to use when submitting the
+        post. For example:
+
+        {'url': 'https://mybucket.s3.amazonaws.com
+         'fields': {'acl': 'public-read',
+                    'key': 'mykey',
+                    'signature': 'mysignature',
+                    'policy': 'mybase64 encoded policy'}
+        }
+    """
+
+    if fields is None:
+        fields = {}
+
+    if conditions is None:
+        conditions = []
+
+    request_signer = client._request_signer
+    serializer = client._serializer
+
+    # We choose the CreateBucket operation model because its url gets
+    # serialized to what a presign post requires.
+    operation_model = client.meta.service_model.operation_model('CreateBucket')
```

Comment:
I considered creating my own model for this. However, I felt ``CreateBucket`` is suitable. All I really need is the url when I serialize the request and the ``CreateBucket`` serialization is exactly how I need the url serialized. Let me know if you want me to change this.